### PR TITLE
Pulumi Helm Deployments: Allow Null Values in Configs

### PR DESCRIPTION
**Pull request by cdibble**
### Proposed changes

I need to be able to pass `ingress.tls=[]` to the Apache Superset helm chart, which won't deploy unless that config is populated, even if it is with a null entry. At this time, null values like that are scrubbed out of helm configs in the `mergeMaps` function, so you can not pass a null value to your helm deployment.

This PR would add a boolean to allow null values through mergeMaps in helm_release.go and expose that option in the SDKs.

### Related issues (optional)

Addresses https://github.com/pulumi/pulumi-kubernetes/issues/2089

### CODE REVIEW

**Comment by cdibble** [comment]
@lblackstone I went ahead and opened this PR after merging it in my fork (https://github.com/cdibble/pulumi-kubernetes/pull/1). Please LMK if you need additional input from me :)

File: CHANGELOG.md
Comment (lblackstone) [inline CHANGELOG.md:13]: This will need to move the the `Unreleased` section

**Review by lblackstone — COMMENTED** [review]
Thanks for the PR! I think the code changes look good, and I've kicked off the CI tests to make sure that's passing. Just one minor update needed on the changelog, and I think this will be ready to merge.

File: CHANGELOG.md
Comment (cdibble) [reply CHANGELOG.md]: > This will need to move the the `Unreleased` section

Oops, misinterpreted the headings. I've made that fix- thanks for the catch :)

**Comment by cdibble** [comment]
> Thanks for the PR! I think the code changes look good, and I've kicked off the CI tests to make sure that's passing. Just one minor update needed on the changelog, and I think this will be ready to merge.

@lblackstone Just resolved that comment- thank you! Please feel free to ping me if you need anything else from me. Thanks for the quick review :)

**Comment by lblackstone** [comment]
@cdibble The CI tests caught an issue: https://github.com/pulumi/pulumi-kubernetes/actions/runs/3342207244/jobs/5534197567#step:20:477

Looks like you'll need to update the function signature in a test.

File: CONTRIBUTING.md
Comment (cdibble) [inline CONTRIBUTING.md:25]: @lblackstone just wanted to point out that I made this change to the documentation. LMK if you'd prefer that I revert that or elaborate on what those commands are doing.

**Comment by cdibble** [comment]
> @cdibble The CI tests caught an issue: https://github.com/pulumi/pulumi-kubernetes/actions/runs/3342207244/jobs/5534197567#step:20:477
> 
> Looks like you'll need to update the function signature in a test.

@lblackstone Ok I just updated that test, hopefully that takes care of it. This does just maintain the previous test coverage, rather than extending it to include coverage of this functionality. I imagine the latter is preferable- I'll take a closer look to see if I can work out how to do that and put another commit on here.

**Comment by cdibble** [comment]
@lblackstone I just pushed a commit that adds one more test with this functionality turned on. We can revert that if it's not desired of course. TBH I am not totally confident I got the patterns right and don't have a dev environment to try it out locally. Let me know if you have any comments :)

File: CONTRIBUTING.md
Comment (lblackstone) [reply CONTRIBUTING.md:25]: LGTM

File: provider/pkg/provider/helm_release_test.go
Comment (lblackstone) [inline provider/pkg/provider/helm_release_test.go:38]: You'll need to add this parameter to the struct definition as well on line 32

Comment (cdibble) [reply provider/pkg/provider/helm_release_test.go:39]: @lblackstone Ok just made that update. Sorry I should've caught that.

**Comment by cdibble** [comment]
FYI I see the error and will try to revise this to get it working. Might take a minute as I think I need to improve my understanding of Go fundamentals and figure out how to run these tests locally.

File: provider/pkg/provider/helm_release.go
Comment (lblackstone) [inline provider/pkg/provider/helm_release.go:1357]:
```suggestion
			if val.IsValid() {
				if val.IsNil() {
					out[k] = nil
				} else {
					out[k] = mapToInterface(v)
				}
			}
```

**Review by lblackstone — CHANGES_REQUESTED** [review]
Ah, I see the problem. I think my suggested change should fix it.

**Comment by cdibble** [comment]
@lblackstone Alrighty- I've enjoyed getting to know Go a bit better and I think that the latest commit should allow these tests to pass with the expected merged map. Please let me know if you have a moment to re-run those CI tests and I'd as always be happy to address any questions or comments.

File: provider/pkg/provider/helm_release.go
Comment (cdibble) [reply provider/pkg/provider/helm_release.go]: @lblackstone Oh great it looks like we ended up in similar places. I'll commit yours as I am certain it is more idiomatic. Thank you!

**Comment by lblackstone** [comment]
> @lblackstone Alrighty- I've enjoyed getting to know Go a bit better and I think that the latest commit should allow these tests to pass with the expected merged map. Please let me know if you have a moment to re-run those CI tests and I'd as always be happy to address any questions or comments.

Thanks for your patience on this. I couldn't figure out why the test was still failing, so I went ahead and debugged it. Turns out that the `IsValid` function returns `false` for `nil` values, so it still wasn't working quite right. I pushed a new commit that passes the tests.

**Comment by cdibble** [comment]
> Thanks for your patience on this. I couldn't figure out why the test was still failing, so I went ahead and debugged it. Turns out that the `IsValid` function returns `false` for `nil` values, so it still wasn't working quite right. I pushed a new commit that passes the tests.

No problem at all. Thanks for your help. Hopefully we're good to go now but please do tag me if this needs any further attention.

**Review by lblackstone — APPROVED** [review]

**Pull request merged**
